src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DefaultTagHelperDescriptorBuilder.cs

```diff
@@ -10,8 +10,31 @@
 
 namespace Microsoft.AspNetCore.Razor.Language;
 
-internal class DefaultTagHelperDescriptorBuilder : TagHelperDescriptorBuilder, IBuilder<TagHelperDescriptor>
+internal partial class DefaultTagHelperDescriptorBuilder : TagHelperDescriptorBuilder, IBuilder<TagHelperDescriptor>
 {
+    private static readonly ObjectPool<DefaultTagHelperDescriptorBuilder> s_pool = DefaultPool.Create(Policy.Instance);
+
+    public static DefaultTagHelperDescriptorBuilder Get(string name, string assemblyName)
+        => Get(TagHelperConventions.DefaultKind, name, assemblyName);
+
+    public static DefaultTagHelperDescriptorBuilder Get(string kind, string name, string assemblyName)
+    {
+        var builder = s_pool.Get();
+
+        builder._kind = kind;
+        builder._name = name;
+        builder._assemblyName = assemblyName;
+
+        // Tells code generation that these tag helpers are compatible with ITagHelper.
+        // For now that's all we support.
+        builder._metadata.Add(TagHelperMetadata.Runtime.Name, TagHelperConventions.DefaultKind);
```

Comment:
Consider extracting this into a common initialization method.

Reply:
Thanks!